### ISSUE TITLE
Best-checkpoint selection uses liver_per_case only with >= tie-break

### ISSUE THREAD

**Issue by NguyenJus** [NguyenJus/pytorch-dense-unet-3d#5]
## Problem

The cascaded driver selects the "best" checkpoint using a single metric with a permissive comparison. Surfaced during the PR #2 code review (deferred as a design decision).

In `dense_unet_3d/training/cascaded_driver.py`, both `run_phase_a` and `run_phase_b` track best via:

```python
val_dice = float(val_metrics.get("liver_per_case", 0.0))
if val_dice >= best_val_dice:
    ...  # save best
```

Two concerns:
1. **Liver-only metric:** tumor performance is never consulted. Combined with the inflated `tumor_per_case` (see #4), a degenerate model that segments liver well but never predicts tumor can be selected as Phase A "best" and reloaded into Phase B (`model.load_state_dict(phase_a_ckpt["model_state_dict"])`).
2. **`>=` tie-break:** a later epoch with an equal score always overwrites the prior best, biasing toward later (not better) weights.

## Files

- `dense_unet_3d/training/cascaded_driver.py` — `run_phase_a`, `run_phase_b` best-tracking + Phase B reload

## Design decision needed

- What metric should drive best-checkpoint selection? (e.g. mean of liver + tumor per-case, a weighted combination, or a presence-aware metric per #4)
- Change `>=` to `>` so ties keep the earlier (already-validated) checkpoint?

Then implement with a test asserting a degenerate tumor-blind model does not win over a balanced one.
